### DISCUSSION TITLE
getStaticProps - Get value from Query paramter url

### DISCUSSION THREAD

**Discussion by mskian** [vercel/next.js#22019] [Help] [answered]
I want to add Get value from the URL

```js
http://localhost:3000/?name=helloworld
```

```js
const pathdata = typeof window !== "undefined" ? window.location.search.split('?name=')[1] : ""
const getdata = encodeURIComponent(pathdata)
console.log(getdata);
export async function getStaticProps() {
  const response = await axios.get(`http://localhost:3000/api/wish/${getdata}`)
  const posts = await response.data
  console.log(posts)
  return {
    props: {
      posts,
    },
  }
}
```
but it ends up with an empty value

**Comment by jamesmosier** [comment, marked as answer]
Hi @mskian. [There is no way to get the query parameter](https://nextjs.org/docs/basic-features/data-fetching#only-runs-at-build-time) in `getStaticProps` because `getStaticProps` runs at build time and therefore has no knowledge of query string params. You could use `next/router` and [`useRouter()`](https://nextjs.org/docs/api-reference/next/router#userouter) to get the query string param at runtime.
